### ISSUE TITLE
ProjectileCreated/Destroyed: use nil instead of -1

### ISSUE THREAD

**Issue by sprunk** [beyond-all-reason/spring#880]
Currently they pass a magic -1 value when no valid value exists. Use nil instead.

**Comment by lhog** [comment]
Yeah, totally makes sense, we need to find users of Recoil's ProjectileCreated/Destroyed, so they can update game side.
